test/unit/webgl/p5.Camera.js

```diff
@@ -0,0 +1,208 @@
+        x: myCam.centerX,
+        y: myCam.centerY,
+        z: myCam.centerZ
+      };
+      for (let i = 0; i < 20; i++) {
+        myCam.pan(1);
+        myCam.pan(-1);
+      }
+      assert.closeTo(myCam.centerX, originalCenterXYZ.x, delta);
+      assert.closeTo(myCam.centerY, originalCenterXYZ.y, delta);
+      assert.closeTo(myCam.centerZ, originalCenterXYZ.z, delta);
+    });
+
+    test('Tilt() back and forth should result in similar centerXYZ', function() {
+      var originalCenterXYZ = {
+        x: myCam.centerX,
+        y: myCam.centerY,
+        z: myCam.centerZ
+      };
+      for (let i = 0; i < 20; i++) {
+        myCam.tilt(1);
+        myCam.tilt(-1);
+      }
+
+      assert.closeTo(myCam.centerX, originalCenterXYZ.x, delta);
+      assert.closeTo(myCam.centerY, originalCenterXYZ.y, delta);
+      assert.closeTo(myCam.centerZ, originalCenterXYZ.z, delta);
+    });
+  });
+
+  suite('Position / Orientation', function() {
+    test('Camera() sets eye, center, and up XYZ properties', function() {
+      myCam.camera(1, 1, 1, 2, 2, 2, 0, 0, 1);
+      assert.strictEqual(myCam.eyeX, 1);
+      assert.strictEqual(myCam.eyeY, 1);
+      assert.strictEqual(myCam.eyeZ, 1);
+      assert.strictEqual(myCam.centerX, 2);
+      assert.strictEqual(myCam.centerY, 2);
+      assert.strictEqual(myCam.centerZ, 2);
+      assert.strictEqual(myCam.upX, 0);
+      assert.strictEqual(myCam.upY, 0);
+      assert.strictEqual(myCam.upZ, 1);
+    });
+
+    test('LookAt() should set centerXYZ', function() {
+      myCam.lookAt(23, 23, 23);
+      assert.strictEqual(myCam.centerX, 23);
+      assert.strictEqual(myCam.centerY, 23);
+      assert.strictEqual(myCam.centerZ, 23);
+    });
+
+    test('Move() back and forth should result in similar eyeXYZ', function() {
+      var originalEyeXYZ = {
+        x: myCam.eyeX,
+        y: myCam.eyeY,
+        z: myCam.eyeZ
+      };
+
+      for (let i = 0; i < 20; i++) {
+        var myVec = {
+          x: Math.random(),
+          y: Math.random(),
+          z: Math.random()
+        };
+        myCam.move(myVec.x, myVec.y, myVec.z);
+        myCam.move(-1 * myVec.x, -1 * myVec.y, -1 * myVec.z);
+      }
+      assert.closeTo(myCam.eyeX, originalEyeXYZ.x, delta);
+      assert.closeTo(myCam.eyeY, originalEyeXYZ.y, delta);
+      assert.closeTo(myCam.eyeZ, originalEyeXYZ.z, delta);
+    });
+
+    test('SetPosition() should set eyeXYZ', function() {
+      var pos = Math.random();
+
+      myCam.setPosition(pos, pos, pos);
+
+      assert.strictEqual(myCam.eyeX, pos);
+      assert.strictEqual(myCam.eyeY, pos);
+      assert.strictEqual(myCam.eyeZ, pos);
+    });
+  });
+
+  suite('Projection', function() {
+    test('ortho() sets renderer uPMatrix', function() {
+      myCam.ortho(-10, 10, -10, 10, 0, 100);
+
+      for (let i = 0; i < myCam.projMatrix.mat4.length; i++) {
+        assert.strictEqual(
+          myCam.projMatrix.mat4[i],
+          myp5._renderer.uPMatrix.mat4[i]
+        );
+      }
```

Comment:
in the same test, we should check if the expected values are in the matrix with additional assertions. you can add an argument to the assert call to print a more helpful error so it's obvious if the error was that the matrices didn't match or that they didn't have the expected values (applies to next test, too)